react-form-controls/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useState } from 'react';
+
+export default function RegistrationFormControlled() {
+  const [userName, setUserName] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(e: any) {
+    // Prevent the browser from reloading the page
+    e.preventDefault();
+
+    // Read the form data
```

Comment:
Since you already have `username` and `password` as state variables, you don't need to read them from the form, you can just use those values directly.